### PR TITLE
add new buttons and make them optional

### PR DESCRIPTION
**Pull request by maidis**
This changes add two new buttons for pastes:

- Download
- Copy to Clipboard

and make all new buttons optional because they don't have any vital functionality.

I don't know if it's okay to write such a long JavaScript code in a single line. Perhaps one of the following would be better:

- JS script/function in HTML
- a separate JS file

I'm also not sure if this is the best way to make something optional. Please let me know if there is a better way. Maybe a new header file can be added for this type variables or maybe a simple admin page can be made.

### CODE REVIEW

File: src/root.cpp [hunk 1]
```diff
@@ -37,6 +37,11 @@
 
 using namespace Cutelyst;
 
+// Some variables for optional features
+const bool socialMediaButtons = 1;
```
Comment (dantti) [inline src/root.cpp]: Use member variables instead of global ones

File: src/root.cpp [hunk 2]
```diff
@@ -55,6 +60,10 @@ void Root::index(Context *c)
 
 void Root::item(Context *c, const QString &uuid)
 {
+    c->setStash(QStringLiteral("social"), socialMediaButtons);
```
Comment (dantti) [inline src/root.cpp]: load the optional settings from c->config()

File: src/root.cpp [hunk 3]
```diff
@@ -60,6 +55,11 @@ void Root::index(Context *c)
 
 void Root::item(Context *c, const QString &uuid)
 {
+    // Some variables for optional features
+    bool socialMediaButtons = c->config(QStringLiteral("social"), true).toBool();
```
Comment (dantti) [inline src/root.cpp]: Reimplemente Controller's:

    virtual bool preFork(Application *app);

And initialize the option as member variables like:

    private:
    bool m_downloadButton;

Comment (maidis) [reply src/root.cpp]: I found example usage of preFork() at Cloudlyst:

- https://github.com/cutelyst/Cloudlyst/blob/2cc1bbd1dcdf48d15916def7daaf75c4c5095d60/src/webdav.h#L82
- https://github.com/cutelyst/Cloudlyst/blob/b27c7a7b1621bc57bab0f51b1905efcf5f7510f0/src/webdav.cpp#L636

But the compilation doesn't succeed when I make changes according to this. And when I try to adapt the variables are not effective.

https://paste.ubuntu.com/p/TfB63tD6kK/

**Comment by dantti** [comment]
you need to include Cutelyst::Application in cpp file

**Comment by maidis** [comment]
Thanks for help. Could you check the code again?

File: src/root.h [hunk 4]
```diff
@@ -69,6 +71,9 @@ class Root : public Controller
 private:
     QElapsedTimer m_cleaupTimer;
     HtmlHighlighter *m_htmlHighlighter;
+    bool m_socialMediaButtons = 0;
```
Comment (dantti) [inline src/root.h]: you should either initialize to true or false, or since you will initilize in prefork just don't initialize.

**Pull request merged**